packages/block-library/src/template-part/variations.js

```diff
@@ -1,56 +1,60 @@
 /**
  * WordPress dependencies
  */
-import { footer, header } from '@wordpress/icons';
 import { store as coreDataStore } from '@wordpress/core-data';
-import { select } from '@wordpress/data';
-import { __ } from '@wordpress/i18n';
+import { store as editorStore } from '@wordpress/editor';
+import { store as blocksStore } from '@wordpress/blocks';
+import { dispatch, select, subscribe } from '@wordpress/data';
 
-const variations = [
-	{
-		name: 'header',
-		title: __( 'Header' ),
-		description: __(
-			"The header template defines a page area that typically contains a title, logo, and main navigation. Since it's a global element it can be present across all pages and posts."
-		),
-		icon: header,
-		attributes: { area: 'header' },
-		scope: [ 'inserter' ],
-	},
-	{
-		name: 'footer',
-		title: __( 'Footer' ),
-		description: __(
-			"The footer template defines a page area that typically contains site credits, social links, or any other combination of blocks. Since it's a global element it can be present across all pages and posts."
-		),
-		icon: footer,
-		attributes: { area: 'footer' },
-		scope: [ 'inserter' ],
-	},
-];
+const unsubscribe = subscribe( () => {
```

Comment:
Here is where I feel there is inconsistency with the naming of the file and the handling of variations. I believe your approach should align with the dynamic registration of the block variations and is related to this issue: https://github.com/WordPress/gutenberg/issues/30739.

Reply:
>  inconsistency with the naming of the file and the handling of variations.

The file name is `variations` and it defines how the variations should be applied.  This seems like a consistent go-to file to find the variations code.

> your approach should align with the dynamic registration of the block variations and is related to this issue: #30739.

🤔  If I am understanding correctly, the current approach would be to define the variations in php AND use a hook in js to get `isActive` to work properly.  So instead of a developer easily finding all the variation code for the block consolidated in a logical place (variations.js), the variations code would be split between the index.php and some other hooks.js file to map the isActive callbacks?  That just seems a bit ugly, and shouldn't variations code fall into a consistent location for every block (variations.js) wherever possible?

After reading that issue, it seems the solution that is moving forward for allowing `isActive` to work from the server registration side is to allow an array for matching block attributes.  This wouldn't allow us to clean up our use case for template parts since the variations are determined in a more complex manner than block attributes and require us to select the entity from the data store for comparison.

So with all that considered... is it preferable to split the variations definitions into two less standard locations rather than keep it consolidated in variations.js?